src/setup.rs

```diff
@@ -41,3 +45,28 @@ fn get_env_port() -> u16 {
 pub fn get_socket_addr() -> SocketAddrV4 {
     SocketAddrV4::new(Ipv4Addr::new(0, 0, 0, 0), get_env_port())
 }
+
+#[derive(Clone)]
+pub struct AppState {
+    connection_pool: Pool<Postgres>,
+}
+
+pub async fn create_app_state() -> AppState {
+    AppState {
+        connection_pool: database::get_connection_pool().await,
```

Comment:
We need this public so that we can access the thing outside of this module.
```suggestion
        pub connection_pool: database::get_connection_pool().await,
```